Unit-01/04-flask-crud/app.py

```diff
@@ -1,3 +1,60 @@
-from flask import Flask
+from flask import Flask, render_template, request, redirect, url_for, abort
+from snack import Snack
+from functions import find_snack
+from flask_modus import Modus
 
-snack_list = []
\ No newline at end of file
+app = Flask(__name__)
+modus = Modus(app)
+
+snacks = [Snack('kit-kat', 'unhealthy')]
+
+@app.route("/")
+def root():
+	return redirect(url_for('index'))
+
+#To read snacks (GET /snacks)
+#To create snack (POST /snacks)
+@app.route("/snacks", methods=["GET", "POST"])
+def index():
+	if(request.method == "POST"):
+		snacks.append(Snack(request.form.get("name"), request.form.get("kind")))
+		return redirect(url_for('index'))
+	return render_template("index.html", snacks=snacks)
+
+#To read form to create snack(GET /snacks/new)
+@app.route("/snacks/new")
+def new():
+	return render_template("new.html")
+
+#To read snack (GET /snacks/<int:id>)
+#To edit snack (PUT /snacks/<int:id>)
+#To delete snack (DELETE /snacks/<int:id>)
+@app.route("/snacks/<int:id>", methods=["GET", "PATCH", "DELETE"])
+def show(id):
+	found = find_snack(id, snacks)
+	if found == None:
+		abort(404)
```

Comment:
two other ways to do this check that are more preferable:

```python
# first way
if not found:
    abort(404)

# second way
if found is None:
    abort(404)
```
the first one is probably better, unless you want to explicitly check for equality to `None`, in which case `is None` is generally the more Pythonic way to do it